client/src/connection_cache.rs

```diff
@@ -160,7 +172,8 @@ impl ConnectionCacheStats {
 }
 
 struct ConnectionMap {
-    map: BTreeMap<SocketAddr, Connection>,
+    map: HashMap<SocketAddr, Connection>,
+    list_of_peers: Vec<SocketAddr>,
```

Comment:
Seems like a job for [`IndexMap`](https://docs.rs/indexmap/latest/indexmap/index.html) ?

Reply:
Yes, that's perfect. I'll update the code.